tbot/main.py

```diff
@@ -57,11 +60,26 @@ def _import_hightlighter() -> typing.Callable[[str], str]:
 # }}}
 
 
+class TbotArgumentParser(argparse.ArgumentParser):
+    def convert_arg_line_to_args(self, arg_line: str) -> List[str]:
+        """
+        Make it possible to use shell variables also in argumentsfiles.
+        """
+        try:
+            arg_line_expanded = string.Template(arg_line).substitute(os.environ)
+        except KeyError as e:
+            raise tbot.error.TbotException(
+                f"Could not find environment variable: {e.args[0]!r}"
+            ) from None
+
+        return arg_line_expanded.split()
```

Comment:
Can you drop the split here as we discussed, then I will apply your patch :)

Reply:
Just tried to drop and get this error (also added this to tbot/newbot.py):

hs@asuslap:tbot-rahix  [allow-shell-vars-in-argumentfiles] $ pre-commit run --all-files
black....................................................................Passed
check for added large files..............................................Passed
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check that executables have shebangs.....................................Passed
check for merge conflicts................................................Passed
debug statements (python)................................................Passed
flake8...................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

tbot/newbot.py:191: error: Return type "str" of "convert_arg_line_to_args" incompatible with return type "List[str]" in supertype "ArgumentParser"
tbot/main.py:63: error: Return type "str" of "convert_arg_line_to_args" incompatible with return type "List[str]" in supertype "ArgumentParser"
Found 2 errors in 2 files (checked 97 source files)

Reply:
Hmm, maybe that gets us closer to the problem:  Does a

```python
        return [arg_line_expanded]
```

fix the `Exception` you were getting?

Reply:
indeed, this fixes it! Send soon a new version with this change